app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt

```diff
@@ -74,18 +75,30 @@ class ExternalPlayer(
 
     @JavascriptInterface
     fun initPlayer(args: String) {
-        val playOptions = PlayOptions.fromJson(JSONObject(args)) ?: return
+        val playOptions = PlayOptions.fromJson(JSONObject(args))
+        val itemId = playOptions?.run { mediaSourceId ?: ids.firstOrNull() }
+        if (playOptions == null || itemId == null) {
+            context.toast(R.string.player_error_invalid_play_options)
+            return
+        }
 
         coroutinesScope.launch {
-            val jellyfinMediaSource = mediaSourceResolver.resolveMediaSource(
-                itemId = playOptions.mediaSourceId!!,
+            mediaSourceResolver.resolveMediaSource(
+                itemId = itemId,
                 deviceProfile = externalPlayerProfile,
                 startTimeTicks = playOptions.startPositionTicks,
                 audioStreamIndex = playOptions.audioStreamIndex,
                 subtitleStreamIndex = playOptions.subtitleStreamIndex,
-            ) ?: return@launch
-
-            playMediaSource(playOptions, jellyfinMediaSource)
+            ).onSuccess { jellyfinMediaSource ->
+                playMediaSource(playOptions, jellyfinMediaSource)
+            }.onFailure { error ->
```

Comment:
Is this an ExoPlayer thing or a Kotlin thing?

Reply:
It's the Result class
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/